### ISSUE TITLE
Relative "quarter" filter doesn't work in SQL queries

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#23258]
**Describe the bug**
When using the "quarter" relative filter on a SQL query via a field filter, then Metabase show "no results". If you do the same query but in a GUI question, everything works as it should

**Logs**
<details>
```
metabase-postgres         | 2022-06-09 16:14:14,957 ERROR middleware.catch-exceptions :: Error processing query: null
metabase-postgres         | {:database_id 2,
metabase-postgres         |  :started_at #t "2022-06-09T16:14:14.852382Z[GMT]",
metabase-postgres         |  :via
metabase-postgres         |  [{:status :failed,
metabase-postgres         |    :class clojure.lang.ExceptionInfo,
metabase-postgres         |    :error "Unable to substitute parameters: null",
metabase-postgres         |    :stacktrace
metabase-postgres         |    ["--> driver.sql.parameters.substitute$substitute$fn__82079.invoke(substitute.clj:86)"
metabase-postgres         |     "driver.sql.parameters.substitute$substitute.invokeStatic(substitute.clj:83)"
metabase-postgres         |     "driver.sql.parameters.substitute$substitute.invoke(substitute.clj:73)"
metabase-postgres         |     "driver.sql$fn__82599$substitute_native_parameters__82592__82604$fn__82608.invoke(sql.clj:46)"
metabase-postgres         |     "driver.sql$fn__82599$substitute_native_parameters__82592__82604.invoke(sql.clj:42)"
metabase-postgres         |     "query_processor.middleware.parameters.native$expand_inner.invokeStatic(native.clj:39)"
metabase-postgres         |     "query_processor.middleware.parameters.native$expand_inner.invoke(native.clj:30)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_one.invokeStatic(parameters.clj:49)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_one.invoke(parameters.clj:40)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_all$replace_49319__49320.invoke(parameters.clj:58)"
metabase-postgres         |     "mbql.util.match.impl$replace_in_collection$iter__20130__20134$fn__20135.invoke(impl.cljc:44)"
metabase-postgres         |     "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:43)"
metabase-postgres         |     "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_all$replace_49319__49320.invoke(parameters.clj:58)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:58)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:52)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:55)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:52)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:76)"
metabase-postgres         |     "query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:72)"
metabase-postgres         |     "query_processor.middleware.parameters$fn__49335$substitute_parameters_STAR___49340$fn__49341.invoke(parameters.clj:81)"
metabase-postgres         |     "query_processor.middleware.parameters$fn__49335$substitute_parameters_STAR___49340.invoke(parameters.clj:78)"
metabase-postgres         |     "query_processor.middleware.parameters$substitute_parameters.invokeStatic(parameters.clj:108)"
metabase-postgres         |     "query_processor.middleware.parameters$substitute_parameters.invoke(parameters.clj:100)"
metabase-postgres         |     "query_processor$preprocess_STAR_$fn__52909.invoke(query_processor.clj:124)"
metabase-postgres         |     "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:122)"
metabase-postgres         |     "query_processor$preprocess_STAR_.invoke(query_processor.clj:117)"
metabase-postgres         |     "query_processor$fn__52917$combined_pre_process__52918$combined_pre_process_STAR___52919.invoke(query_processor.clj:204)"
metabase-postgres         |     "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__51365$fn__51370.invoke(resolve_database_and_driver.clj:35)"
metabase-postgres         |     "driver$do_with_driver.invokeStatic(driver.clj:75)"
metabase-postgres         |     "driver$do_with_driver.invoke(driver.clj:71)"
metabase-postgres         |     "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__51365.invoke(resolve_database_and_driver.clj:34)"
metabase-postgres         |     "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__47470.invoke(fetch_source_query.clj:281)"
metabase-postgres         |     "query_processor.middleware.store$initialize_store$fn__47658$fn__47659.invoke(store.clj:11)"
metabase-postgres         |     "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
metabase-postgres         |     "query_processor.store$do_with_store.invoke(store.clj:38)"
metabase-postgres         |     "query_processor.middleware.store$initialize_store$fn__47658.invoke(store.clj:10)"
metabase-postgres         |     "query_processor.middleware.normalize_query$normalize$fn__51637.invoke(normalize_query.clj:22)"
metabase-postgres         |     "metabase_enterprise.audit_app.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__58302.invoke(handle_audit_queries.clj:130)"
metabase-postgres         |     "query_processor.middleware.constraints$add_default_userland_constraints$fn__48727.invoke(constraints.clj:53)"
metabase-postgres         |     "query_processor.middleware.process_userland_query$process_userland_query$fn__51576.invoke(process_userland_query.clj:145)"
metabase-postgres         |     "query_processor.middleware.catch_exceptions$catch_exceptions$fn__51948.invoke(catch_exceptions.clj:162)"
metabase-postgres         |     "query_processor.reducible$async_qp$qp_STAR___44197$thunk__44199.invoke(reducible.clj:100)"
metabase-postgres         |     "query_processor.reducible$async_qp$qp_STAR___44197$fn__44201.invoke(reducible.clj:105)"],
metabase-postgres         |    :error_type :qp,
metabase-postgres         |    :ex-data
metabase-postgres         |    {:type :qp,
metabase-postgres         |     :params
metabase-postgres         |     {"created_at"
metabase-postgres         |      (metabase.driver.common.parameters/map->FieldFilter
metabase-postgres         |       {:field
metabase-postgres         |        {:description "The date and time an order was submitted.",
metabase-postgres         |         :database_type "timestamp",
metabase-postgres         |         :semantic_type :type/CreationTimestamp,
metabase-postgres         |         :table_id 5,
metabase-postgres         |         :coercion_strategy nil,
metabase-postgres         |         :name "created_at",
metabase-postgres         |         :settings nil,
metabase-postgres         |         :effective_type :type/DateTime,
metabase-postgres         |         :nfc_path nil,
metabase-postgres         |         :parent_id nil,
metabase-postgres         |         :id 42,
metabase-postgres         |         :visibility_type :normal,
metabase-postgres         |         :display_name "Created At",
metabase-postgres         |         :fingerprint
metabase-postgres         |         {:global {:distinct-count 9998, :nil% 0.0},
metabase-postgres         |          :type {:type/DateTime {:earliest "2016-04-30T18:56:13.352Z", :latest "2020-04-19T14:07:15.657Z"}}},
metabase-postgres         |         :base_type :type/DateTime},
metabase-postgres         |        :value {:type :date/all-options, :value "past0quarters", :id "87d6ba33-7cb9-0d85-af64-4065023a4916"}})},
metabase-postgres         |     :parsed-query
metabase-postgres         |     ["SELECT \"public\".\"orders\".\"id\" AS \"id\", \"public\".\"orders\".\"user_id\" AS \"user_id\", \"public\".\"orders\".\"product_id\" AS \"product_id\", \"public\".\"orders\".\"subtotal\" AS \"subtotal\", \"public\".\"orders\".\"tax\" AS \"tax\", \"public\".\"orders\".\"total\" AS \"total\", \"public\".\"orders\".\"discount\" AS \"discount\", \"public\".\"orders\".\"created_at\" AS \"created_at\", \"public\".\"orders\".\"quantity\" AS \"quantity\"\nFROM \"public\".\"orders\"\nWHERE "
metabase-postgres         |      (metabase.driver.common.parameters/->Param "created_at")]}}],
metabase-postgres         |  :error_type :qp,
metabase-postgres         |  :json_query
metabase-postgres         |  {:constraints {:max-results 10000, :max-results-bare-rows 2000},
metabase-postgres         |   :type :native,
metabase-postgres         |   :middleware {:js-int-to-string? true, :ignore-cached-results? false, :process-viz-settings? false},
metabase-postgres         |   :native
metabase-postgres         |   {:template-tags
metabase-postgres         |    {"created_at"
metabase-postgres         |     {:id "87d6ba33-7cb9-0d85-af64-4065023a4916",
metabase-postgres         |      :name "created_at",
metabase-postgres         |      :display-name "Created at",
metabase-postgres         |      :type :dimension,
metabase-postgres         |      :dimension [:field 42 nil],
metabase-postgres         |      :widget-type :date/all-options,
metabase-postgres         |      :default nil}},
metabase-postgres         |    :query
metabase-postgres         |    "SELECT \"public\".\"orders\".\"id\" AS \"id\", \"public\".\"orders\".\"user_id\" AS \"user_id\", \"public\".\"orders\".\"product_id\" AS \"product_id\", \"public\".\"orders\".\"subtotal\" AS \"subtotal\", \"public\".\"orders\".\"tax\" AS \"tax\", \"public\".\"orders\".\"total\" AS \"total\", \"public\".\"orders\".\"discount\" AS \"discount\", \"public\".\"orders\".\"created_at\" AS \"created_at\", \"public\".\"orders\".\"quantity\" AS \"quantity\"\nFROM \"public\".\"orders\"\nWHERE {{created_at}}"},
metabase-postgres         |   :database 2,
metabase-postgres         |   :parameters
metabase-postgres         |   [{:type "date/all-options",
metabase-postgres         |     :value "past0quarters",
metabase-postgres         |     :target ["dimension" ["template-tag" "created_at"]],
metabase-postgres         |     :id "87d6ba33-7cb9-0d85-af64-4065023a4916"}],
metabase-postgres         |   :async? true,
metabase-postgres         |   :cache-ttl nil},
metabase-postgres         |  :status :failed,
metabase-postgres         |  :class java.lang.NullPointerException,
metabase-postgres         |  :stacktrace
metabase-postgres         |  ["--> driver.common.parameters.dates$fn__39804.invokeStatic(dates.clj:172)"
metabase-postgres         |   "driver.common.parameters.dates$fn__39804.invoke(dates.clj:168)"
metabase-postgres         |   "driver.common.parameters.dates$fn__39911$execute_decoders__39916$fn__39917$fn__39919.invoke(dates.clj:320)"
metabase-postgres         |   "driver.common.parameters.dates$fn__39911$execute_decoders__39916$fn__39917.invoke(dates.clj:318)"
metabase-postgres         |   "driver.common.parameters.dates$fn__39911$execute_decoders__39916.invoke(dates.clj:313)"
metabase-postgres         |   "driver.common.parameters.dates$fn__39983$date_string__GT_range__39992$fn__39998.invoke(dates.clj:375)"
metabase-postgres         |   "driver.common.parameters.dates$fn__39983$date_string__GT_range__39992.invoke(dates.clj:351)"
metabase-postgres         |   "driver.common.parameters.dates$fn__39983$date_string__GT_range__39992$fn__39993.invoke(dates.clj:367)"
metabase-postgres         |   "driver.common.parameters.dates$fn__39983$date_string__GT_range__39992.invoke(dates.clj:351)"
metabase-postgres         |   "driver.sql.parameters.substitution$fn__81658$date_range_field_filter__GT_replacement_snippet_info__81663$fn__81664.invoke(substitution.clj:206)"
metabase-postgres         |   "driver.sql.parameters.substitution$fn__81658$date_range_field_filter__GT_replacement_snippet_info__81663.invoke(substitution.clj:204)"
metabase-postgres         |   "driver.sql.parameters.substitution$fn__81811$field_filter__GT_replacement_snippet_info__81816$fn__81822.invoke(substitution.clj:275)"
metabase-postgres         |   "driver.sql.parameters.substitution$fn__81811$field_filter__GT_replacement_snippet_info__81816.invoke(substitution.clj:256)"
metabase-postgres         |   "driver.sql.parameters.substitution$fn__81842.invokeStatic(substitution.clj:301)"
metabase-postgres         |   "driver.sql.parameters.substitution$fn__81842.invoke(substitution.clj:286)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute_field_filter.invokeStatic(substitute.clj:16)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute_field_filter.invoke(substitute.clj:11)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute_param.invokeStatic(substitute.clj:33)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute_param.invoke(substitute.clj:27)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute_STAR_$fn__82070.invoke(substitute.clj:66)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute_STAR_.invokeStatic(substitute.clj:59)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute_STAR_.invoke(substitute.clj:56)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute$fn__82079.invoke(substitute.clj:84)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute.invokeStatic(substitute.clj:83)"
metabase-postgres         |   "driver.sql.parameters.substitute$substitute.invoke(substitute.clj:73)"
metabase-postgres         |   "driver.sql$fn__82599$substitute_native_parameters__82592__82604$fn__82608.invoke(sql.clj:46)"
metabase-postgres         |   "driver.sql$fn__82599$substitute_native_parameters__82592__82604.invoke(sql.clj:42)"
metabase-postgres         |   "query_processor.middleware.parameters.native$expand_inner.invokeStatic(native.clj:39)"
metabase-postgres         |   "query_processor.middleware.parameters.native$expand_inner.invoke(native.clj:30)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_one.invokeStatic(parameters.clj:49)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_one.invoke(parameters.clj:40)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_all$replace_49319__49320.invoke(parameters.clj:58)"
metabase-postgres         |   "mbql.util.match.impl$replace_in_collection$iter__20130__20134$fn__20135.invoke(impl.cljc:44)"
metabase-postgres         |   "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:43)"
metabase-postgres         |   "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_all$replace_49319__49320.invoke(parameters.clj:58)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:58)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:52)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:55)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:52)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:76)"
metabase-postgres         |   "query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:72)"
metabase-postgres         |   "query_processor.middleware.parameters$fn__49335$substitute_parameters_STAR___49340$fn__49341.invoke(parameters.clj:81)"
metabase-postgres         |   "query_processor.middleware.parameters$fn__49335$substitute_parameters_STAR___49340.invoke(parameters.clj:78)"
metabase-postgres         |   "query_processor.middleware.parameters$substitute_parameters.invokeStatic(parameters.clj:108)"
metabase-postgres         |   "query_processor.middleware.parameters$substitute_parameters.invoke(parameters.clj:100)"
metabase-postgres         |   "query_processor$preprocess_STAR_$fn__52909.invoke(query_processor.clj:124)"
metabase-postgres         |   "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:122)"
metabase-postgres         |   "query_processor$preprocess_STAR_.invoke(query_processor.clj:117)"
metabase-postgres         |   "query_processor$fn__52917$combined_pre_process__52918$combined_pre_process_STAR___52919.invoke(query_processor.clj:204)"
metabase-postgres         |   "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__51365$fn__51370.invoke(resolve_database_and_driver.clj:35)"
metabase-postgres         |   "driver$do_with_driver.invokeStatic(driver.clj:75)"
metabase-postgres         |   "driver$do_with_driver.invoke(driver.clj:71)"
metabase-postgres         |   "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__51365.invoke(resolve_database_and_driver.clj:34)"
metabase-postgres         |   "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__47470.invoke(fetch_source_query.clj:281)"
metabase-postgres         |   "query_processor.middleware.store$initialize_store$fn__47658$fn__47659.invoke(store.clj:11)"
metabase-postgres         |   "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
metabase-postgres         |   "query_processor.store$do_with_store.invoke(store.clj:38)"
metabase-postgres         |   "query_processor.middleware.store$initialize_store$fn__47658.invoke(store.clj:10)"
metabase-postgres         |   "query_processor.middleware.normalize_query$normalize$fn__51637.invoke(normalize_query.clj:22)"
metabase-postgres         |   "metabase_enterprise.audit_app.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__58302.invoke(handle_audit_queries.clj:130)"
metabase-postgres         |   "query_processor.middleware.constraints$add_default_userland_constraints$fn__48727.invoke(constraints.clj:53)"
metabase-postgres         |   "query_processor.middleware.process_userland_query$process_userland_query$fn__51576.invoke(process_userland_query.clj:145)"
metabase-postgres         |   "query_processor.middleware.catch_exceptions$catch_exceptions$fn__51948.invoke(catch_exceptions.clj:162)"
metabase-postgres         |   "query_processor.reducible$async_qp$qp_STAR___44197$thunk__44199.invoke(reducible.clj:100)"
metabase-postgres         |   "query_processor.reducible$async_qp$qp_STAR___44197$fn__44201.invoke(reducible.clj:105)"],
metabase-postgres         |  :card_id 8,
metabase-postgres         |  :context :question,
metabase-postgres         |  :error nil,
metabase-postgres         |  :row_count 0,
metabase-postgres         |  :running_time 0,
metabase-postgres         |  :data {:rows [], :cols []}}
metabase-postgres         | 
```
</details>

**To Reproduce**
With the sample database
1. Do a SQL query like the following:
```
SELECT "public"."orders"."id" AS "id", "public"."orders"."created_at" AS "created_at"
FROM "public"."orders"
WHERE {{created_at}}
```
2. Configure {{created_at}} filter as a field filter connected to the Orders->created_at column
3. In the field filter, click on "relative date" and enter "past 30 quarters"
4. You see "no data", and the backend returns an error in the logs

**Expected behavior**
See the data that should be returned

**Screenshots**
NA

**Information about your Metabase Installation:**
- Your browser and the version: Brave latest
- Your operating system: Pop 22.04
- Your databases: Postgres 14.2
- Metabase version: 1.43.2
- Metabase hosting environment: Docker
- Metabase internal database: Postgres 14.2

**Severity**
P2'ish?

**Additional context**
Reported by a customer

**Comment by paoliniluis** [comment]
Derivative of https://github.com/metabase/metabase/issues/20683

**Comment by flamber** [comment]
Making this a P1, since we're showing "No results". Unsure if this is caused by incorrect FE request or if BE is missing something.
Backend log errors with `Unable to substitute parameters: null`

**Comment by alxnddr** [comment]
Looks like a BE issue

**Comment by dpsutton** [comment]
I'm unable to reproduce the "no results" part of this bug. I've seen it happen with @paoliniluis with a docker setup but just running the jar locally gives me:
<img width="1310" alt="image" src="https://user-images.githubusercontent.com/6377293/173667841-548750f3-2a88-4f30-a10b-996450ecedd1.png">

I'm going to continue trying to recreate the issue but there are two issues at play here:
- error in query is silent and comes back as no errors
- queries do not support `last30quarters`, `thisquarter`, `next30quarters`

**Comment by dpsutton** [comment]
I have identified the issue. On java  java 14+ a null pointer exception has a lovely message about what was nil ([JEP-358](https://openjdk.org/jeps/358) ) vs no message prior:

```
;; java 11
jshell> HashSet x = null;
x ==> null

jshell> x.contains(3);
|  Exception java.lang.NullPointerException
|        at (#2:1)

jshell>

;; java 17
jshell> HashSet x = null;
x ==> null

jshell> x.contains(3);
|  Exception java.lang.NullPointerException: Cannot invoke "java.util.HashSet.contains(Object)" because "REPL.$JShell$11.x" is null
|        at (#4:1)
```

and more germane to us, in Clojure:
```clojure
;; java 17
catch-exceptions=> (let [x nil] (x 3))
Execution error (NullPointerException) at metabase.query-processor.middleware.catch-exceptions/eval130385 (REPL:41).
Cannot invoke "clojure.lang.IFn.invoke(Object)" because "x" is null

;; java 11

catch-exceptions=> (let [x nil] (x 3))
Execution error (NullPointerException) at metabase.driver.sql.parameters.substitution/eval118508 (REPL:17).
null
```

It seems that the frontend considers a query result an error if and only if the `"error"` is a non-nil string:

```javascript
// objects are edited for clarity/brevity:
// from java-11
{
  "error_type": "qp",
  "status": "failed",
  "class": "class java.lang.NullPointerException",
  "stacktrace": [...],
  "error": null,
}

// vs java-17
{
  "error_type": "qp",
  "status": "failed",
  "class": "class java.lang.NullPointerException",
  "stacktrace": [...],
  "error": "Cannot invoke \"clojure.lang.IFn.invoke(Object)\" because \"to_period\" is null",,
}
```

this seems to also be the cause for some of the logged messages we have seen:
```
;; java 11
ERROR middleware.catch-exceptions :: Error processing query: null

;; java 17
ERROR middleware.catch-exceptions :: Error processing query: Cannot invoke "clojure.lang.IFn.invoke(Object)" because "to_period" is null
```
Except there is a bug in this as well:
```clojure
(let [formatted-exception (format-exception* query e @extra-info)]
  ;; should be `(:error formatted-exception)`, `format-exception` is a multimethod
  (log/error (str (trs "Error processing query: {0}" (:error format-exception))
                  "\n" (u/pprint-to-str formatted-exception)))
  (qp.context/resultf formatted-exception context))
```

### What should happen

The frontend needs to be better about recognizing when an error has occurred. I'm reading through the code and almost all of the properties on this are optional. This is obviously a case where `:error` can be missing; `:via` is only present when exceptions have causes; `:error_type` is a property we set in certain places but probably not all; `:status` can be "failed" or "interrupted" and hard to know if those cases are the same or not; ex-data is not top level. My thinking is to add a new `{"is_exception" true ...}` at the top level that is always present if the catch exception middleware was triggered. We can rely on this value, regardless of if there are multiple causes of an exception, message or not, etc.

**Comment by alxnddr** [comment]
@dpsutton can we make `error` non-nullable? For instance, we can use a generic error message when there is no specific exception message. Since we did not have issues like that before it seems like it was always present and that's why FE relies on it. We already have multiple error fields and adding a new one might be an overkill. What do you think?

I believe this issue has exactly the same root cause https://github.com/metabase/metabase/issues/23281

**Comment by dpsutton** [comment]
@alxnddr we can do that. It's quite simple to do I think which is a nice win. But i'm always hesitant to use presence of strings for logic rather than data. But i'll put up a patch with that approach.

**Comment by alxnddr** [comment]
@dpsutton sounds great, thank you! This string won't be like a flag just for logic, we render the value of `error` in the UI. So the BE decides what users should know about certain errors

**Comment by dpsutton** [comment]
So this bug is half solved with the merging of #23346 . Quarter filters will still error (they have always been erroring) but it will not show "no results" but correctly show that there was an error in the query. I'm going to implement them now so hopefully both bugs are fixed for 43.4

**Comment by dpsutton** [comment]
Will be in 43.4 with backport PR: https://github.com/metabase/metabase/pull/23399
